### PR TITLE
Update Airflow version to 2.5.2

### PR DESCRIPTION
**Pull request by pierrejeambrun**
Airflow 2.5.2 has been released.

### CODE REVIEW

**Review by potiuk — APPROVED** [review]

File: RELEASE_NOTES.rst [hunk 1]
```diff
@@ -21,6 +21,114 @@
 
 .. towncrier release notes start
 
+Airflow 2.5.2 (2023-03-13)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst:24]:
```suggestion
Airflow 2.5.2 (2023-03-14)
```

Or maybe even 15 now?

**Review by jedcunningham — APPROVED** [review]

File: RELEASE_NOTES.rst (hunk 1 above)
Comment (pierrejeambrun) [reply RELEASE_NOTES.rst:24]: Yes 15 is good, I'll follow up with the airflow-site update and the EndOfLife PR once the appropriate release date is set.

Fixed.

Comment (pierrejeambrun) [reply RELEASE_NOTES.rst:24]: I got additional instructions from Ephraim, this needs to be in a separate PR for easy cherry picking on v2-5-test. Updating

Comment (jedcunningham) [reply RELEASE_NOTES.rst:24]: Oh, yes, good call.

**Pull request merged**